cloud_governance/policy/policy_operations/aws/zombie_cluster/delete_ec2_resources.py

```diff
@@ -372,6 +372,21 @@ def __delete_network_interface(self, resource_id: str):
         """
         try:
             resource_list = self.ec2_operations.get_network_interface()
+
+            # Check if this is a cluster resource before proceeding
+            network_interface_data = self.__get_cluster_references(resource_id=resource_id, resource_list=resource_list,
+                                                                   input_resource_id='NetworkInterfaceId',
+                                                                   output_result='')
+            if network_interface_data:
+                network_interface = network_interface_data[0]
+                tags = network_interface.get('TagSet', [])
+                if self.cluster_tag and not self.__is_cluster_resource(tags, self.cluster_tag):
+                    logger.info(f'Network interface {resource_id} is not a cluster resource. Skipping deletion.')
```

Comment:
I think we should ask for positive and NOT negative:
if self.cluster_tag and self.__is_cluster_resource(tags, self.cluster_tag):
No need to print log for every resource that is not related to cluster, we just want to skip it.

Reply:
I meant that we should check if only when self.__is_cluster_resource(tags, self.cluster_tag).

`if self.cluster_tag and self.__is_cluster_resource(tags, self.cluster_tag):`

in other words, if it not a cluster resource filter it out.